expan/core/early_stopping.py

```diff
@@ -0,0 +1,228 @@
+import os
+from os.path import dirname, join, realpath
+import numpy as np
+from scipy.stats import gaussian_kde, norm, cauchy
+from pystan import StanModel
+
+import expan.core.statistics as statx
+
+__location__ = realpath(join(os.getcwd(), dirname(__file__)))
+
+def obrien_fleming(information_fraction, alpha=0.05):
+	"""
+	Calculate an approximation of the O'Brien-Fleming alpha spending function.
+
+	Args:
+		information_fraction: share of the information amount at the point 
+			of evaluation, e.g. the share of the maximum sample size
+		alpha: type-I error rate
+
+	Returns:
+		float: redistributed alpha value at the time point with the given 
+			   information fraction
+	"""
+	return (1-norm.cdf(norm.ppf(1-alpha/2)/np.sqrt(information_fraction)))*2
+
+
+def group_sequential(x, 
+					 y, 
+					 spending_function='obrien_fleming', 
+					 information_fraction=1, 
+					 alpha=0.05, 
+					 cap=8):
+	"""
+	Group sequential method to determine whether to stop early or not.
+
+	Args:
+		x (array_like): sample of a treatment group
+		y (array_like): sample of a control group
+		spending_function: name of the alpha spending function, currently
```

Comment:
that can be the spending function itself instead of the string with its name, see `callable(spending_function)` call below